### PR TITLE
python312Packages.ufal-chu-liu-edmonds: init at 1.0.3

### PR DESCRIPTION
**Pull request by ViZiD**
## Description of changes
https://github.com/ufal/chu_liu_edmonds/tree/master
Chu-Liu-Edmonds maximum spanning algorithm from TurboParser for use within Python

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [x] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/development/python-modules/ufal-chu-liu-edmonds/default.nix [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    inherit version;
+    hash = "sha256-v3tv6cYWoFPPgaO6KXR2uUk3MsZ458Tz5wKeFW8fzNE=";
+  };
+
+  build-system = [
+    setuptools
+    wheel
+  ];
+
+  doCheck = false;
```
Comment (Moraxyc) [inline pkgs/development/python-modules/ufal-chu-liu-edmonds/default.nix:26]:
```suggestion
  nativeCheckInputs = [ numpy ];

  checkPhase = ''
    runHook preCheck

    cd tests
    python test.py

    runHook postCheck
  '';
```

File: pkgs/development/python-modules/ufal-chu-liu-edmonds/default.nix [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+  src = fetchPypi {
+    pname = "ufal.chu_liu_edmonds";
+    inherit version;
+    hash = "sha256-v3tv6cYWoFPPgaO6KXR2uUk3MsZ458Tz5wKeFW8fzNE=";
+  };
+
+  build-system = [
+    setuptools
+    wheel
```
Comment (Moraxyc) [inline pkgs/development/python-modules/ufal-chu-liu-edmonds/default.nix:23]:
```suggestion
```

Doesn't seem to be needed.

**Review by Moraxyc — APPROVED** [review]
LGTM, thanks

**Comment by ViZiD** [comment]
@Aleksanaa

File: pkgs/development/python-modules/ufal-chu-liu-edmonds/default.nix [hunk 3]
```diff
@@ -0,0 +1,41 @@
+{
+  lib,
+  buildPythonPackage,
+  fetchPypi,
+
```
Comment (Aleksanaa) [inline pkgs/development/python-modules/ufal-chu-liu-edmonds/default.nix:5]:
```suggestion
```

File: pkgs/development/python-modules/ufal-chu-liu-edmonds/default.nix [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+  setuptools,
+
```
Comment (Aleksanaa) [inline pkgs/development/python-modules/ufal-chu-liu-edmonds/default.nix:7]:
```suggestion
```

**Review by Aleksanaa — COMMENTED** [review]
Others lgtm

**Review by azahi — APPROVED** [review]

**Pull request merged**